.github/workflows/web-pr.yml

```diff
@@ -0,0 +1,59 @@
+name: Build and Test web app
+
+on: [ pull_request ]
+
+jobs:
+  test:
+    name: Test web app
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+
+      - name: Setup JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Load secret files
+        run: |
+          echo "keystore.file=pr.jks" >> ./mobile/signing.properties.prod
+          echo "keystore.password=123456" >> ./mobile/signing.properties.prod
+          echo "alias.name=pr" >> ./mobile/signing.properties.prod
+          echo "alias.password=123456" >> ./mobile/signing.properties.prod
+
+      - name: Install Firebase Tools and Emulators
+        run: sudo npm install -g firebase-tools
+
+      - name: Run tests
+        env:
+          GOOGLE_SERVICE_ACCOUNT: ${{ secrets.FIREBASE_GOOGLE_SERVICE_ACCOUNT }}
+          GOOGLE_DATABASE_NAME: ${{ secrets.FIREBASE_GOOGLE_DATABASE_NAME }}
+          GOOGLE_BUCKET_NAME: ${{ secrets.FIREBASE_GOOGLE_BUCKET_NAME }}
+          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+        run: |
+          chmod +x ../.github/scripts/emulator-env.sh
+          ../.github/scripts/emulator-env.sh > ../.github/scripts/run.sh
+          chmod +x ../.github/scripts/run.sh
+          sudo firebase emulators:exec --only firestore --project tupperdate-developement 'bash ../.github/scripts/run.sh'
```

Comment:
I'm not sure what you are trying to do, but if you want to include a script from another, you should use the command `source` or in bash `.`. So in `run.sh` add a line:
```bash
. ../.github/scripts/emulator-env.sh
```

And then you should have access to all functions and variables, etc. In addition your `emulator-env.sh` does not need to be `chmod +x`